pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java

```diff
@@ -372,7 +375,9 @@ public synchronized int getPGType(String pgTypeName) throws SQLException {
     ResultSet rs = oidStatement.getResultSet();
     if (rs.next()) {
       oid = (int) rs.getLong(1);
-      _oidToPgName.put(oid, pgTypeName);
+      String internalName = rs.getString(2);
+      _oidToPgName.put(oid, internalName);
+      _pgNameToOid.put(internalName, oid);
     }
     _pgNameToOid.put(pgTypeName, oid);
```

Comment:
>The order of that is irelevant?

Whatever it means, code that has `x = 3; x = 4;` must not exists.
It should be either `x = 3;` or `x = 4;`, not both of them at the same time.